### ISSUE TITLE
[Story] Object geometry primitives in the editor

### ISSUE THREAD

**Issue by HexaField** [XRFoundation/XREngine#6776]
It would be great to be able to throw primitive shapes into the scene in as easily as other objects, and even able to draw objects. Things like cubes, spheres, planes, etc

**Comment by SYBIOTE** [comment]
for the primitives i can just let it be a model component, and add the respective shape/geometry on it in three js when the component is added to the scene 

now i have a question 
do we create separate components for all the shapes , or just a single component, with a  dropdown for the shapes?

**Comment by SYBIOTE** [comment]
we might want to rework the ground plane component into this primitive component

**Comment by SYBIOTE** [comment]
questions, 

how do we handle csg for more than two entites interacting?
union and intersect can be done on multiple, what is the expected behavior for difference on multiple entities?
for the csg we could follow different UX
1. for multiple entity interaction , we could follow what godot does
- the csg component is the parent entity
- entities with model under it form a group 
- entities with csg component under it, treat the group as a whole model
and so on 

2. csg is constrained to two entities at a time

**Comment by HexaField** [comment]
> we might want to rework the ground plane component into this primitive component 

No, ground planes are not solid geometries and have special logic associated.

**Comment by HexaField** [comment]
> for the primitives i can just let it be a model component, and add the respective shape/geometry on it in three js when the component is added to the scene 
> 
> now i have a question 
> do we create separate components for all the shapes , or just a single component, with a  dropdown for the shapes?
> 
> 

It should just be the model component, and we have a special menu to add primitives. No need to add components for each.

**Comment by SYBIOTE** [comment]
got that, lets start with that

**Comment by SYBIOTE** [comment]
done with the geometry, its going to be a while before we can focus on the other and advanced features

**Comment by SYBIOTE** [comment]
I'm ready to integrate csg when needed
